src/tamp_improv/approaches/improvisational/policies/mpc.py

```diff
@@ -62,51 +71,60 @@ def train(
             shape = (self.config.horizon,) + action_shape
             self._last_solution = np.zeros(shape, dtype=np.float32)
 
-    def get_action(self, obs: ObsType_contra) -> ActType_co:
-        """Get action using predictive sampling."""
+    def get_action(self, obs: ObsType) -> ActType:
+        """Get action using MPC."""
         if self._env is None:
-            raise ValueError("Policy not initialized - call train() first")
-        return cast(ActType_co, self._solve(obs))
+            raise ValueError("Policy not initialized")
+        return cast(ActType, self._solve(obs))
```

Comment:
are you assuming that actions are always NDArray?

Reply:
I'm not sure how to deal with this. In your mlrp_course repo you would return `Trajectory[TrajOptAction]` from `_solve`, but since we decide not to define such a complicated thing as `Trajectory`, we have to deal with an `NDArray` specifically...? Please let me know if you have any suggestions on how to make this more general.

Reply:
one idea would be to implement an `array_to_action` function in each environment and then use that to do the conversion here. but since our actions are `NDArray`s for now I think this is fine -- if you want to be a little safer though, you could add an assertion like `assert isinstance(action, np.ndarray)`. (`cast` would just silently fail)